pkg/dataobj/metastore/object.go

```diff
@@ -144,6 +144,16 @@ func matchersToString(matchers []*labels.Matcher) string {
 	return s.String()
 }
 
+func (m *ObjectMetastore) ResolveStrategy(tenants []string) ResolveStrategyType {
+	slices.Sort(tenants)
+	slices.Sort(m.cfg.EnabledTenantIDs)
+
+	if m.cfg.IndexStoragePrefix != "" && slices.Equal(m.cfg.EnabledTenantIDs, tenants) {
```

Comment:
Do you mean slices.Contains for each tenant here?
If we set two tenants in EnabledTenantIDs, I think this check would fail as there would only ever be 1 tenant in the query ctx